src/python/pants/backend/python/goals/run_python_requirement.py

```diff
@@ -0,0 +1,181 @@
+    dependencies: PythonRequirementDependenciesField
+    modules: PythonRequirementModulesField
+    resolve: PythonRequirementResolveField
+    entry_point: PexEntryPointField
+
+
+@memoized
+def _invert_module_mapping(
+    resolve: ResolveName, module_mapping: ThirdPartyPythonModuleMapping
+) -> InvertedModuleMapping:
+    """Provide an inverted module mapping that can specify the set of modules known to be fulfilled
+    by a given target address."""
+    d: dict[Address, list[str]] = defaultdict(list)
+    modules_to_providers = module_mapping.resolves_to_modules_to_providers.get(resolve)
+    if not modules_to_providers:
+        return FrozenDict()
+
+    for module_name, providers in modules_to_providers.items():
+        for provider in providers:
+            d[provider.addr].append(module_name)
+
+    return FrozenDict((address, tuple(modules)) for address, modules in d.items())
+
+
+@rule_helper
+async def _resolve_entry_point(
+    module_mapping: InvertedModuleMapping, field_set: PythonRequirementFieldSet
+) -> EntryPoint:
+
+    modules = field_set.modules.value
+    reqs = field_set.requirements.value
+    entry_point_raw = field_set.entry_point.value
+
+    if entry_point_raw:
+        resolved_entry_point = await Get(
+            ResolvedPexEntryPoint,
+            ResolvePexEntryPointRequest(field_set.entry_point),
+        )
+        entry_point = resolved_entry_point.val
+
+        if not entry_point:
+            raise Exception("The provided `entry_point` was not valid.")
+
+        return entry_point
+
+    entry_point_module: str
+    if modules and len(modules) == 1:
+        # Unambiguous module specified in the `BUILD` file
+        entry_point_module = modules[0]
+    elif len(module_mapping.get(field_set.address, ())) == 1:
+        # Check the third-party module mapping
+        entry_point_module = module_mapping[field_set.address][0]
+    elif len(reqs) == 1:
+        # Use the canonicalized project name for a single-requirement target
+        entry_point_module = canonicalize_project_name(reqs[0].project_name)
+    else:
+        raise Exception(
+            "Requirement must provide a single module, specify a single requirement, or specify "
+            "an `entry_point`"
+        )
+
+    return EntryPoint(entry_point_module)
+
+
+@rule(level=LogLevel.DEBUG)
+async def create_python_requirement_run_request(
+    field_set: PythonRequirementFieldSet,
+    pex_env: PexEnvironment,
+    python_setup: PythonSetup,
+    module_mapping: ThirdPartyPythonModuleMapping,
+) -> RunRequest:
+
+    addresses = [field_set.address]
+
+    resolve = field_set.resolve.value
+    if not resolve:
+        resolve = python_setup.default_resolve
+
+    modules_for_address = _invert_module_mapping(resolve, module_mapping)
+    entry_point = await _resolve_entry_point(modules_for_address, field_set)
+    filename = entry_point.spec.replace(".", "__").replace(":", "___")
+
+    pex_request = await Get(
+        PexRequest,
```

Comment:
You can actually just request `VenvPex` at this point, there are rules which already pass that exact pex env